velox/functions/prestosql/tests/DecimalArithmeticTest.cpp

```diff
@@ -291,6 +291,14 @@ TEST_F(DecimalArithmeticTest, multiply) {
                   HugeInt::build(0x08FFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF)},
               DECIMAL(38, 0))}),
       "Decimal overflow. Value '119630519620642428561342635425231011830' is not in the range of Decimal Type");
+
+  // The sum of input scales exceeds result precision.
+  VELOX_ASSERT_THROW(
+      evaluate(
+          "c0 * c0",
+          makeRowVector(
+              {makeFlatVector<int128_t>({1000, 2000}, DECIMAL(38, 30))})),
+      "");
```

Comment:
Would you add an error message here?

Reply:
Got it. I will follow up this change. Thanks.

Reply:
Hi @mbasmanova, I tried to do a refactor by making getType return a status, which carries the error message. Here is my current change https://github.com/rui-mo/velox/commit/d00da6acdc0652ee1efd9603d595afe2099ca2e5#diff-cfd002b96c5cd427ceb98cb16fd13986a90ba3514ec66f5a330009a9978381f8R1143-R1165.
> It would be nice to add an error message to resultType check in SimpleFunctionRegistry::resolveFunction.

To pass the message through, several functions need to be refactored to return status as
`getType -> tryResolveType -> tryResolveReturnType -> resolveFunction`.
And the callers need to be changed correspondingly. Do you think the idea makes sense? If so, I will continue to work on it.

Reply:
@rui-mo Rui, sorry I wasn't very clear. 

> // TODO Perhaps, modify getType to add suppress-errors flag.

I'm not sure if this TODO makes sense. I also don't quite understand why we have try-catch in SignatureBinder in the first place. The change to have getType return Status seems quite invasive, but it doesn't help with this particular code because we still return nullptr. Unless this is blocking, perhaps, we could continue make process on decimal support in the fuzzer: https://github.com/facebookincubator/velox/pull/9149 What do you think?

Reply:
@mbasmanova Thanks. It makes sense to me. I will prioritize the work in the fuzzer.